ams/server/src/main/java/com/netease/arctic/server/ArcticServiceContainer.java

```diff
@@ -301,57 +302,72 @@ private class ConfigurationHelper {
     private JSONObject yamlConfig;
 
     public void init() throws IOException {
-      initServiceConfig();
+      Map<String, Object> envConfig = initEnvConfig();
+      initServiceConfig(envConfig);
       initContainerConfig();
     }
 
     @SuppressWarnings("unchecked")
-    private void initServiceConfig() throws IOException {
+    private void initServiceConfig(Map<String, Object> envConfig) throws IOException {
       LOG.info("initializing service configuration...");
       String configPath = Environments.getConfigPath() + "/" + SERVER_CONFIG_FILENAME;
       LOG.info("load config from path: {}", configPath);
       yamlConfig = new JSONObject(new Yaml().loadAs(Files.newInputStream(Paths.get(configPath)), Map.class));
       JSONObject systemConfig = yamlConfig.getJSONObject(ArcticManagementConf.SYSTEM_CONFIG);
-      Map<String, Object> expandedConfigurationMap = Maps.newHashMap();
-      expandConfigMap(systemConfig, "", expandedConfigurationMap);
-      validateConfig(expandedConfigurationMap);
-      serviceConfig = Configurations.fromObjectMap(expandedConfigurationMap);
+      expandConfigMap(systemConfig, "", envConfig);
+      validateConfig(envConfig);
+      serviceConfig = Configurations.fromObjectMap(envConfig);
       SqlSessionFactoryProvider.getInstance().init(serviceConfig);
     }
 
+    private Map<String, Object> initEnvConfig() {
+      LOG.info("initializing system env configuration...");
+      Map<String, String> envConfigMap = System.getenv().entrySet()
+              .stream()
+              .filter(entry -> entry.getKey().startsWith(ArcticManagementConf.SYSTEM_CONFIG))
```

Comment:
The value of `SYSTEM_CONFIG` is `ams`. Should we use a new env such as `AMORO_CONF_` ?

Reply:
Does this mean that we need to extract all environment variables starting with AMORO_CONF_ ？

Reply:
Referring to the environment variable configuration methods of other open source software, such as 1, 2, and 3.

Considering the common practice of using uppercase letters and underscores in environment variables, I suggest:

    Using "AMORO_" as the prefix for environment variable configuration.
    Using underscores instead of dots.
    Keeping the configuration consistent with the config for ease of configuration.

For example, use `AMORO_http-server_bind-port` to represent the configuration item `ams.http-server.bind-port`.

@zhoujinsong  WDYT ?

Reply:
Here are some references:

- https://etcd.io/docs/v3.4/op-guide/configuration/
- https://github.com/apache/spark/blob/master/conf/spark-env.sh.template
- https://github.com/Gradiant/dockerized-hadoop/blob/master/README.md